File: Problems/SelfCheckout_10/SelfCheckout.py
taxRate = 0.055;
addItem = "yes"
moreItems = True
itemCount = 1
itemList = {}
while moreItems:
	item = input("Enter the price of item %i: " % itemCount)
	quantity = input("Enter the quantity of item %i: " % itemCount)
	if item.isnumeric() and quantity.isnumeric():
		itemList[item] = quantity
		addItem = input("Add another item? ")
		while True:
			if addItem.lower() == "yes" or addItem.lower() == "y":
				itemCount += 1
				break
			if addItem.lower() == "no" or addItem.lower() == "n":
				moreItems = False
				break
			else:
				print("Please enter yes or no")
				continue
	else:
		print("Please enter two positive numbers")

subtotal = 0

for item, quantity in itemList.items():
	subtotal += float(item) * int(quantity)
	
tax = subtotal * taxRate
total = subtotal + tax

print("Subtotal: %f" % (subtotal))
print("Taxes: %f" % (tax))
print("Total: %f" % (total))


#TO DO: Check if isnumeric() works for floats

File: Problems/SelfCheckout_10/SelfCheckout.cpp
#include <iostream>
#include <string>
#include <stdlib.h>
#include <math.h>
#include <map>

using namespace std;

int main()
{
    const double taxRate = 0.055;
    double subtotal, total, tax;
    string addItem = "yes", item, quantity;
	bool moreItems = true;
	int itemCount = 1;
	map<string, string> itemList;

	while(moreItems)
	{
		cout << "Enter the price of item " << itemCount << " : ";
		cin >> item;
		cout << "Enter the quantity of item " << itemCount << " : ";
		cin >> quantity;
		
		itemList.insert(pair<string, string>(item, quantity));
		cout << "Add another item? ";
		cin >> addItem;
		if(addItem == "no")
			moreItems = false;
		else
		{
			itemCount++;
		}
		/*while(true)
		{
			if(addItem.)
			{
			}
			else if()
			{
			}
			else
			{
			}
		}*/
	}
	
	for(map<string, string>::const_iterator it = itemList.begin(); it != itemList.end(); ++it)
	{
		subtotal += atof(it->first.c_str()) * atof(it->second.c_str());
	}

	tax = subtotal * taxRate;
	total = tax + subtotal;

	cout << "Subtotal: " << subtotal << "\n";
	cout << "Taxes: " << tax << "\n";
	cout << "Total: " << total << "\n";
}

/*TO DO: Check if input is valid and program the add another item flow*/
